flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -234,16 +230,15 @@ class SubtaskCheckpointCoordinatorImpl implements SubtaskCheckpointCoordinator {
 
     private static ChannelStateWriter openChannelStateWriter(
             String taskName,
-            CheckpointStorage checkpointStorage,
-            Environment env,
-            int maxSubtasksPerChannelStateFile) {
+            CheckpointStorageWorkerView checkpointStorageWorkerView,
+            Environment env) {
         return new ChannelStateWriterImpl(
                 env.getJobVertexId(),
                 taskName,
                 env.getTaskInfo().getIndexOfThisSubtask(),
-                checkpointStorage,
+                checkpointStorageWorkerView,
                 env.getChannelStateExecutorFactory(),
-                maxSubtasksPerChannelStateFile);
+                1 /* one logical channel state file/executor per subtask */);
```

Comment:
If we hard code `maxSubtasksPerChannelStateFile` is 1, it means `execution.checkpointing.unaligned.max-subtasks-per-channel-state-file` doesn't take effect  after this PR, right?

I'm thinking could we keep the `execution.checkpointing.unaligned.max-subtasks-per-channel-state-file` first? and deprecate it after `unified file merging mechanism` is stable.

WDYT?

Reply:
> If we hard code maxSubtasksPerChannelStateFile is 1, it means execution.checkpointing.unaligned.max-subtasks-per-channel-state-file doesn't take effect after this PR, right?

Right.

In fact, both merge mechanisms can work simultaneously. The `unified file merging mechanism` allows different output streams to be written to a physical file, and the current merging mechanism allows multiple subtasks to be written to one output stream. 

I will keep `execution.checkpointing.unaligned.max-subtasks-per-channel-state-file` first.
- If `execution.checkpointing.unaligned.max-subtasks-per-channel-state-file:1` and `state.checkpoints.file-merging.enabled: true`, only the `unified file merging mechanism` takes effect.
- if `execution.checkpointing.unaligned.max-subtasks-per-channel-state-file > 1` and `state.checkpoints.file-merging.enabled: false`, only the current mechanism takes effect.
-  if `execution.checkpointing.unaligned.max-subtasks-per-channel-state-file > 1` and `state.checkpoints.file-merging.enabled: true`, both two mechanism take effect.